[Dataset Quality] Apply chunking algorithm for getIndexBasicStats

rStelmach opened the pull request:
After adding a chunking algorithm for getIndexBasicStats errors are not appearing anymore. 
For testing purposes I created 600 indicies using this command : 
`for i in {1..600}; do
  curl -u <username>:<password> -X POST "<elasticsearch_url>/<index_base_name>-$i/_doc" -H 'Content-Type: application/json' -d '{"@timestamp": "<timestamp>", "message": "<message>"}'
done`

 and created this test for testing the telemetry function :

```

import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
import { Client } from '@elastic/elasticsearch';
import { lastValueFrom } from 'rxjs';
import {
  getIndexBasicStats,
  addMappingsToIndices,
  getIndicesInfoForPattern,
} from '@kbn/dataset-quality-plugin/server/services/data_telemetry/helpers';

export type IndicesGetResponse = Record<IndexName, any>;
export type IndexName = string;

async function runTest() {
  const client = new Client({
    node: 'http://localhost:9200',
    auth: { username: '', password: '' },
  });

  const results = {
    addMappings: await testAddMappingsToIndicesTypeSafety(client),
    indicesInfo: await testGetIndicesInfoForPattern(client),
    basicStats: await testGetIndexBasicStats(client),
  };

  process.stdout.write('Test results:\n');
  process.stdout.write(`- addMappingsToIndices: ${results.addMappings ? 'SUCCESS' : 'FAILED'}\n`);
  process.stdout.write(
    `- getIndicesInfoForPattern: ${results.indicesInfo ? 'SUCCESS' : 'FAILED'}\n`
  );
  process.stdout.write(`- getIndexBasicStats: ${results.basicStats ? 'SUCCESS' : 'FAILED'}\n`);
}

async function testAddMappingsToIndicesTypeSafety(client: Client): Promise<boolean> {
  try {
    const logsIndexPatterns = [{ patternName: 'logs-*', shipper: 'test', pattern: 'logs-*' }];
    const dataStreamsInfo = [
      {
        patternName: 'logs-*',
        shipper: 'test',
        isDataStream: false,
        name: 'logs-test-1',
        indices: ['logs-test-1'],
        mapping: undefined,
      },
    ];
    const observable = addMappingsToIndices({
      esClient: client as unknown as ElasticsearchClient,
      dataStreamsInfo,
      logsIndexPatterns,
    });
    const result = await lastValueFrom(observable);

    const hasMapping = result.length > 0 && !!result[0].mapping;
    process.stdout.write(
      `addMappingsToIndices ${hasMapping ? 'found mappings' : 'no mapping found'}\n`
    );
    if (hasMapping) {
      const keys = Object.keys(result[0].mapping!);
      process.stdout.write(`  Keys: ${keys.join(', ')}\n`);
    }
    return true;
  } catch (error) {
    process.stdout.write(
      `addMappingsToIndices error: ${error instanceof Error ? error.message : String(error)}\n`
    );
    return false;
  }
}

async function testGetIndicesInfoForPattern(client: Client): Promise<boolean> {
  try {
    const pattern = { patternName: 'logs-*', shipper: 'test', pattern: 'logs-*' };
    const result = await getIndicesInfoForPattern({
      esClient: client as unknown as ElasticsearchClient,
      pattern,
    });

    if (result.length > 0) {
      process.stdout.write(
        `getIndicesInfoForPattern found ${result.length} indices, sample: ${result[0].name}\n`
      );
      return true;
    } else {
      process.stdout.write('getIndicesInfoForPattern returned no results\n');
      return false;
    }
  } catch (error) {
    process.stdout.write(
      `getIndicesInfoForPattern error: ${error instanceof Error ? error.message : String(error)}\n`
    );
    return false;
  }
}

async function testGetIndexBasicStats(client: Client): Promise<boolean> {
  try {
    const response = await client.cat.indices({
      format: 'json',
      h: 'index',
      index: 'logs-*',
    });

    const indicesList = response as Array<{ index: string }>;

    if (indicesList.length === 0) {
      process.stdout.write('No indices found to test with\n');
      return false;
    }

    const indexBasicInfos = indicesList.map((indexObj) => ({
      patternName: 'logs-*',
      shipper: 'test',
      isDataStream: false,
      name: indexObj.index,
      indices: [indexObj.index],
      mapping: undefined,
    }));

    process.stdout.write(`Testing with ${indexBasicInfos.length} indices\n`);

    const observable = getIndexBasicStats({
      esClient: client as unknown as ElasticsearchClient,
      indices: indexBasicInfos,
      breatheDelay: 0,
    });

    const result = await lastValueFrom(observable);
    process.stdout.write(`getIndexBasicStats processed ${result.length} indices\n`);
    return result.length > 0;
  } catch (error) {
    process.stdout.write(
      `getIndexBasicStats error: ${error instanceof Error ? error.message : String(error)}\n`
    );
    return false;
  }
}

runTest().catch((err) => {
  process.stdout.write('Unhandled error in test:', err);
});
      
```

<!--ONMERGE {"backportTargets":["8.19"]} ONMERGE-->

rStelmach committed:
apply chunking algorythm for getIndexBasicStats and getFailureStoreStats

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

rStelmach committed:
bring back old version of getFailureStoreStats

rStelmach committed:
add reduceAsyncChunks for addMappingsToIndices and getIndicesInfoForPattern

rStelmach committed:
add error handling

rStelmach committed:
remove unused export

rStelmach commented:
@mohamedhamed-ahmed  
I updated few more functions and added some small error handling( I couldn't think of a better one). Let me know what you think

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/302092)
* Commit: ce6d86f6beeec580e73672690fdca509e4a481dd

### Failed CI Steps
* [FTR Configs #11](https://buildkite.com/elastic/kibana-pull-request/builds/302092#0196fde8-a24b-49e0-9bd6-d524913d225f)
* [FTR Configs #11](https://buildkite.com/elastic/kibana-pull-request/builds/302092#0196fe10-b2b6-42d3-a54f-1c22216890be)
* [FTR Configs #28](https://buildkite.com/elastic/kibana-pull-request/builds/302092#0196fde8-a256-4dcc-8338-67842ecbc1bd)
* [FTR Configs #39](https://buildkite.com/elastic/kibana-pull-request/builds/302092#0196fde8-a25d-4f5d-bfa9-9fb2d6e6b367)
* [FTR Configs #90](https://buildkite.com/elastic/kibana-pull-request/builds/302092#0196fde8-a281-47f4-b8e8-a3babdab307c)
* [FTR Configs #120](https://buildkite.com/elastic/kibana-pull-request/builds/302092#0196fde8-a298-4ae0-8268-7c1eab57e481)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/302092#0196fde8-a24b-49e0-9bd6-d524913d225f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/302092/jobs/0196fde8-a24b-49e0-9bd6-d524913d225f/artifacts/0196fe10-8243-48b9-b39c-fcf075206d89) FTR Configs #<span></span>11 / GenAI - Knowledge Base Entries APIs @<span></span>ess Security AI Assistant - Indices with `semantic_text` fields "before all" hook for "should return all existing indices with `semantic_text` fields"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/302092#0196fe10-b2b6-42d3-a54f-1c22216890be) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/302092/jobs/0196fe10-b2b6-42d3-a54f-1c22216890be/artifacts/0196fe18-5da6-4674-8bee-97ca931edd54) FTR Configs #<span></span>11 / GenAI - Knowledge Base Entries APIs @<span></span>ess Security AI Assistant - Indices with `semantic_text` fields "before all" hook for "should return all existing indices with `semantic_text` fields"

### History
* :green_heart: [Build #301708](https://buildkite.com/elastic/kibana-pull-request/builds/301708) succeeded 08dfe3616cb418072c620a2a10a75eab06c481e7
* :broken_heart: [Build #301449](https://buildkite.com/elastic/kibana-pull-request/builds/301449) failed be79c5cd3348d7f4227d88d0d5133620e8892e8f

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302092","number":302092,"commit":"ce6d86f6beeec580e73672690fdca509e4a481dd"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301708","number":301708,"commit":"08dfe3616cb418072c620a2a10a75eab06c481e7"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301449","number":301449,"commit":"be79c5cd3348d7f4227d88d0d5133620e8892e8f"}],"number":302092}
buildkite-pr-comment-->

rStelmach committed:
Merge branch 'main' into 214817-dataset-quality-telemetry-background-tasks-generates-too-big-requests-and-responses

awahab07 reviewed (changes requested):
Great to see the refactor to optimize ES calls and handling the output for errors.

Left some comments as certain error handlings may overfit and suppress some errors, which may result in data inconsistencies.


About where to focus what optimization and how to move forward, I suggest:
- [`addMappingsToIndices`](https://github.com/elastic/kibana/blob/b4767bb108fd1325f4e78c5cedc9980fc8586a08/x-pack/platform/plugins/shared/dataset_quality/server/services/data_telemetry/helpers.ts#L99) could potentially result in errors where the output is too large and less likely to cause `too_long_http_line_exception`. So filtering the output is crucial in this function (chunking is still good here though).

- [`getIndexBasicStats`](https://github.com/elastic/kibana/blob/b4767bb108fd1325f4e78c5cedc9980fc8586a08/x-pack/platform/plugins/shared/dataset_quality/server/services/data_telemetry/helpers.ts#L225) looks like the main culprit causing `too_long_http_line_exception` errors. So request chunking, filtering output and handling large output errors is important here.

- To ensure the applied filtering is stripping required fields, it'd be great to have a scenario to test the Before and After effect to make sure the resultant stats are intact. For this, a mix of data streams and regular indices with names conforming to [log patterns](https://github.com/elastic/kibana/blob/b4767bb108fd1325f4e78c5cedc9980fc8586a08/x-pack/platform/plugins/shared/dataset_quality/server/services/data_telemetry/constants.ts#L27) can be created, while [collected fields](https://github.com/elastic/kibana/blob/b4767bb108fd1325f4e78c5cedc9980fc8586a08/x-pack/platform/plugins/shared/dataset_quality/server/services/data_telemetry/constants.ts#L79) being ingested. Also a few documents in the failure store for a few log patterns present.

rStelmach committed:
adjust error handling and optimize telemetry data collection to prevent ES exceptions

rStelmach committed:
Merge branch 'main' into 214817-dataset-quality-telemetry-background-tasks-generates-too-big-requests-and-responses

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/303096)
* Commit: b71847c39749b8d058aaeb29f2f0ed0ebd36f621

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #302092](https://buildkite.com/elastic/kibana-pull-request/builds/302092) failed ce6d86f6beeec580e73672690fdca509e4a481dd
* :green_heart: [Build #301708](https://buildkite.com/elastic/kibana-pull-request/builds/301708) succeeded 08dfe3616cb418072c620a2a10a75eab06c481e7
* :broken_heart: [Build #301449](https://buildkite.com/elastic/kibana-pull-request/builds/301449) failed be79c5cd3348d7f4227d88d0d5133620e8892e8f

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303096","number":303096,"commit":"b71847c39749b8d058aaeb29f2f0ed0ebd36f621"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302092","number":302092,"commit":"ce6d86f6beeec580e73672690fdca509e4a481dd"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301708","number":301708,"commit":"08dfe3616cb418072c620a2a10a75eab06c481e7"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301449","number":301449,"commit":"be79c5cd3348d7f4227d88d0d5133620e8892e8f"}],"number":303096}
buildkite-pr-comment-->

rStelmach committed:
Merge branch 'main' into 214817-dataset-quality-telemetry-background-tasks-generates-too-big-requests-and-responses

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/capture_oas_snapshot   --include-path /api/status   --include-path /api/alerting/rule/   --include-path /api/alerting/rules   --include-path /api/actions   --include-path /api/security/role   --include-path /api/spaces   --include-path /api/streams   --include-path /api/fleet   --include-path /api/dashboards   --include-path /api/saved_objects/_import   --include-path /api/saved_objects/_export   --include-path /api/maintenance_window --update'

kibanamachine committed:
[CI] Auto-commit changed files from 'make api-docs'

awahab07 commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into 214817-dataset-quality-telemetry-background-tasks-generates-too-big-requests-and-responses

awahab07 committed:
Update usages of `reduceAsyncChunks` to `processAsyncInChunks`.

awahab07 committed:
Remove explicit typing.

awahab07 reviewed (approved):
Thanks for the changes.

I've tested the generated telemetry events payload on my end and it looks identical before and after the PR commits. 

<table>
<tr><th>Before</th><th>After</th></tr>
<tr>
<td>

```json
[
    {
        "beat": [
        ],
        "doc_count": 16,
        "failure_store_doc_count": 0,
        "failure_store_index_count": 0,
        "field_count": 2,
        "field_existence": {
            "@timestamp": 16
        },
        "index_count": 1,
        "managed_by": [
        ],
        "namespace_count": 0,
        "package_name": [
        ],
        "pattern_name": "filebeat",
        "shipper": "filebeat",
        "size_in_bytes": 18021,
        "structure_level": {
            "1": 16
        }
    },
    {
        "beat": [
        ],
        "doc_count": 1440,
        "failure_store_doc_count": 0,
        "failure_store_index_count": 0,
        "field_count": 79,
        "field_existence": {
            "@timestamp": 1440,
            "cloud.provider": 1440,
            "container.id": 1440,
            "data_stream.dataset": 1440,
            "data_stream.namespace": 1440,
            "data_stream.type": 1440,
            "event.dataset": 1440,
            "host.name": 1440,
            "kubernetes.pod.uid": 1440,
            "log.level": 1440,
            "message": 1440,
            "service.name": 1440
        },
        "index_count": 1,
        "managed_by": [
        ],
        "namespace_count": 1,
        "package_name": [
        ],
        "pattern_name": "apache",
        "size_in_bytes": 2940697,
        "structure_level": {
            "6": 1440
        }
    },
    {
        "beat": [
        ],
        "doc_count": 55606,
        "failure_store_doc_count": 0,
        "failure_store_index_count": 0,
        "field_count": 105,
        "field_existence": {
            "@timestamp": 55606,
            "cloud.provider": 5760,
            "container.id": 5760,
            "data_stream.dataset": 55606,
            "data_stream.namespace": 55606,
            "data_stream.type": 55606,
            "event.dataset": 5760,
            "host.name": 55606,
            "kubernetes.pod.uid": 5760,
            "log.level": 55600,
            "message": 5760,
            "service.name": 55600
        },
        "index_count": 8,
        "managed_by": [
        ],
        "namespace_count": 1,
        "package_name": [
        ],
        "pattern_name": "dsns-logs",
        "size_in_bytes": 26329771,
        "structure_level": {
            "2": 49846,
            "6": 5760
        }
    }
]
```

</td>

<td>

```json
[
    {
        "beat": [
        ],
        "doc_count": 16,
        "failure_store_doc_count": 0,
        "failure_store_index_count": 0,
        "field_count": 2,
        "field_existence": {
            "@timestamp": 16
        },
        "index_count": 1,
        "managed_by": [
        ],
        "namespace_count": 0,
        "package_name": [
        ],
        "pattern_name": "filebeat",
        "shipper": "filebeat",
        "size_in_bytes": 18021,
        "structure_level": {
            "1": 16
        }
    },
    {
        "beat": [
        ],
        "doc_count": 1440,
        "failure_store_doc_count": 0,
        "failure_store_index_count": 0,
        "field_count": 79,
        "field_existence": {
            "@timestamp": 1440,
            "cloud.provider": 1440,
            "container.id": 1440,
            "data_stream.dataset": 1440,
            "data_stream.namespace": 1440,
            "data_stream.type": 1440,
            "event.dataset": 1440,
            "host.name": 1440,
            "kubernetes.pod.uid": 1440,
            "log.level": 1440,
            "message": 1440,
            "service.name": 1440
        },
        "index_count": 1,
        "managed_by": [
        ],
        "namespace_count": 1,
        "package_name": [
        ],
        "pattern_name": "apache",
        "size_in_bytes": 2940697,
        "structure_level": {
            "6": 1440
        }
    },
    {
        "beat": [
        ],
        "doc_count": 55606,
        "failure_store_doc_count": 0,
        "failure_store_index_count": 0,
        "field_count": 105,
        "field_existence": {
            "@timestamp": 55606,
            "cloud.provider": 5760,
            "container.id": 5760,
            "data_stream.dataset": 55606,
            "data_stream.namespace": 55606,
            "data_stream.type": 55606,
            "event.dataset": 5760,
            "host.name": 55606,
            "kubernetes.pod.uid": 5760,
            "log.level": 55600,
            "message": 5760,
            "service.name": 55600
        },
        "index_count": 8,
        "managed_by": [
        ],
        "namespace_count": 1,
        "package_name": [
        ],
        "pattern_name": "dsns-logs",
        "size_in_bytes": 26329771,
        "structure_level": {
            "2": 49846,
            "6": 5760
        }
    }
]
```

</td>
</tr>
</table>


The test data was generated via:

```bash
node scripts/synthtrace logs_traces_hosts --from=now-1w --to=now \
--scenarioOpts='{
  "ingestHosts": true,
  "ingestTraces": true,
  "numSpaces": 3,
  "datasets": [
    "mysql.error", "mysql.slowlog",
    "kubernetes.audit_logs", "kubernetes.container_logs",
    "nginx.access", "nginx.error",
    "system.auth", "system.syslog",
    "apache.error", "apache.access",
    "postgresql.error", "postgresql.log",
    "redis.error", "redis.log",
    "mongodb.error", "mongodb.log",
    "haproxy.error", "haproxy.log",
    "activemq.audit", "activemq.log",
    "elasticsearch.audit", "elasticsearch.slowlog",
    "elasticsearch.server", "elasticsearch.deprecation", "elasticsearch.gc",
    "akamai.siem",
    "auditd.log", "auditd_manager.auditd",
    "cloud_security_posture.findings", "cloud_security_posture.vulnerabilities",
    "docker.container_logs",
    "elastic_agent.apm_server", "elastic_agent.filebeat", "elastic_agent.heartbeat",
    "fleet_server.output_health", "fleet_server.logs",
    "kibana.audit", "kibana.log",
    "microsoft_sqlserver.audit", "microsoft_sqlserver.log",
    "network_traffic.http", "network_traffic.dns"
  ]
}'
```
The dataset names could be changed however to match all the the collected [patterns](https://github.com/elastic/kibana/blob/b4767bb108fd1325f4e78c5cedc9980fc8586a08/x-pack/platform/plugins/shared/dataset_quality/server/services/data_telemetry/constants.ts#L27) for a more robust _before_ and _after_ comparison.

rStelmach committed:
Merge branch 'main' into 214817-dataset-quality-telemetry-background-tasks-generates-too-big-requests-and-responses

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/307282)
* Commit: 27d0d829594834777586879241d7e6c708bd5eb7

### Failed CI Steps
* [FTR Configs #47](https://buildkite.com/elastic/kibana-pull-request/builds/307282#01975e1f-c8ed-4f48-a756-6dbcfa163df9)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/307282#01975e1f-c8ed-4f48-a756-6dbcfa163df9) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/307282/jobs/01975e1f-c8ed-4f48-a756-6dbcfa163df9/artifacts/01975e2b-5c43-4c49-bcb8-2b89f19d12fb) FTR Configs #<span></span>47 / alerting api integration security and spaces enabled - Group 2 Alerting and Actions Telemetry test telemetry should retrieve telemetry data in the expected format

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #307152](https://buildkite.com/elastic/kibana-pull-request/builds/307152) failed 552db687aa363872c1da97248a7ea44aac1caf98
* :broken_heart: [Build #304707](https://buildkite.com/elastic/kibana-pull-request/builds/304707) failed 0198acb735a88eafcf294afb5593c837b183f402
* :green_heart: [Build #303096](https://buildkite.com/elastic/kibana-pull-request/builds/303096) succeeded b71847c39749b8d058aaeb29f2f0ed0ebd36f621
* :broken_heart: [Build #302092](https://buildkite.com/elastic/kibana-pull-request/builds/302092) failed ce6d86f6beeec580e73672690fdca509e4a481dd
* :green_heart: [Build #301708](https://buildkite.com/elastic/kibana-pull-request/builds/301708) succeeded 08dfe3616cb418072c620a2a10a75eab06c481e7

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307282","number":307282,"commit":"27d0d829594834777586879241d7e6c708bd5eb7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307152","number":307152,"commit":"552db687aa363872c1da97248a7ea44aac1caf98"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/304707","number":304707,"commit":"0198acb735a88eafcf294afb5593c837b183f402"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303096","number":303096,"commit":"b71847c39749b8d058aaeb29f2f0ed0ebd36f621"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302092","number":302092,"commit":"ce6d86f6beeec580e73672690fdca509e4a481dd"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301708","number":301708,"commit":"08dfe3616cb418072c620a2a10a75eab06c481e7"}],"number":307282}
buildkite-pr-comment-->

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.19|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/223380">](https://github.com/elastic/kibana/pull/223380)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->
